### PR TITLE
Introduce mypy type checking

### PR DESCRIPTION
**Pull request by hukkin**
Closes https://github.com/zksync-sdk/zksync-python/issues/13

Fixes the `make mypy` and `tox -e mypy` invocations, which used to run unittests, now run mypy.
Fixes all mypy errors.

### CODE REVIEW

**Comment by hukkin** [comment]
Hey @Deniallugo is mypy something that you want in this project? I'm asking because it took some time and effort to make it pass initially, and now after the latest changes there are pretty big merge conflicts (and mypy errors after fixing those), which I don't want to get into fixing if this isn't something you want merged.

**Comment by popzxc** [comment]
My 5 cents is that `mypy` is pretty important and it's better to have it. The final decision is up to @Deniallugo though.

**Comment by Deniallugo** [comment]
Sorry, for the big gap. 
I think, that mypy is really important for this product. 
So, please use it everywhere

**Review by Deniallugo — APPROVED** [review]

**Comment by hukkin** [comment]
@Deniallugo :+1: I fixed all the conflicts

File: zksync_sdk/wallet.py
Comment (StanislavBreadless) [inline zksync_sdk/wallet.py:106]: The optional should've remained here, right? It seems like  a part of the typing introduced by the PR

Comment (vyastrebovvareger) [reply zksync_sdk/wallet.py:106]: I should work like even without it. But yes, you are right. How can I checkout to this branch and fix it by the commit ?

Comment (StanislavBreadless) [reply zksync_sdk/wallet.py:106]: I guess the best way is to start the new branch which preserves the history with @hukkin's commits and push it separately

Comment (vyastrebovvareger) [reply zksync_sdk/wallet.py:106]: Do you mean checkout the : hukkin/zksync-python repo create the separated branch inside the master and do the new PR for that rigtht?

Comment (hukkin) [reply zksync_sdk/wallet.py]: I've fixed mypy in the current branch for you

Comment (hukkin) [reply zksync_sdk/wallet.py]: Implicit optional is not PEP484 compliant and mypy will error on it in the future: https://github.com/python/mypy/issues/9091

Comment (vyastrebovvareger) [reply zksync_sdk/wallet.py]: But It's not all. I'm doing the fixes now. there is only 1 question: Can i push into this repo separated branch, that's it?

Comment (hukkin) [reply zksync_sdk/wallet.py]: If you have write access to zksync-sdk/zksync-python you can also push to my branch

Comment (vyastrebovvareger) [reply zksync_sdk/wallet.py]: There must be 1 more thing: 
transfer, eth_signature = await self.build_forced_exit(target, `token_obj`, fee_int 
instead of just token

Comment (hukkin) [reply zksync_sdk/wallet.py]: yeah that's fixed in the commit I made

**Comment by vyastrebovvareger** [comment]
I do not know but it looks like the some library interface has been changes/updated. 

from eth_account.signers.base import BaseAccount now accepts only SignableMessage type. Let looking for fix

**Comment by hukkin** [comment]
Fixed the mypy error. It's not a change in the API. It's just that `eth_account` didn't distribute type data before the latest version so mypy was not as strict here before.

**Comment by vyastrebovvareger** [comment]
I got the same issue with the test_is_public_key_onset. Currently I can't get the point why it's failed

**Comment by vyastrebovvareger** [comment]
Locally it's working fine

**Comment by vyastrebovvareger** [comment]
fail of `test_is_public_key_onset` occurs randomly but as far as I can see it depends on timeout and execution flow.

**Pull request merged**
